templates/popular.html

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Popular Movies</title>
+    <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='styles.css') }}" />
+</head>
+<body>
+    <div class="navigation">
+        <a href="{{ url_for('popular') }}">Popular</a>
+        <a href="{{ url_for('content_based_html') }}">Content Based</a>
+        <a href="{{ url_for('collaborative_html') }}">Collaborative</a>
+    </div>
+    <div class="search-results">
+        <h1>Popular Movies</h1>
+    </div>
+    <div class="movie-cards" id="movie-cards">
+        {% for movie in movies %}
```

Comment:
The {% for %} loop is not properly closed. It should be closed with {% endfor %}.